File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/logic/levelling/strategy/impl/variable/VariableLevellingStrategy.java
package io.github.arcaneplugins.levelledmobs.bukkit.logic.levelling.strategy.impl.variable;

import io.github.arcaneplugins.levelledmobs.bukkit.logic.levelling.strategy.LevellingStrategy;
import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context;
import javax.annotation.Nonnull;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.spongepowered.configurate.CommentedConfigurationNode;

public final class VariableLevellingStrategy extends LevellingStrategy {

    // TODO private final String placeholder;

    public VariableLevellingStrategy(
        final int minLevel,
        final int maxLevel
    ) {
        super("Variable", minLevel, maxLevel);
        //TODO
        throw new IllegalStateException("Not implemented");
    }

    @Override
    public @Nullable Integer generate(
        @NotNull final Context context
    ) {
        //TODO
        throw new IllegalStateException("Not implemented");
    }

    @Override
    public @NotNull String replaceInFormula(
        @NotNull final String formula,
        @NotNull final Context context
    ) {
        //TODO
        throw new IllegalStateException("Not implemented");
    }

    public static @Nonnull VariableLevellingStrategy parse(
        @Nonnull final CommentedConfigurationNode node
    ) {
        //TODO
        throw new IllegalStateException("Not implemented");
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/logic/levelling/strategy/impl/variable/VariableLevellingStrategy.kt
package io.github.arcaneplugins.levelledmobs.bukkit.logic.levelling.strategy.impl.variable

import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context
import io.github.arcaneplugins.levelledmobs.bukkit.logic.levelling.strategy.LevellingStrategy
import org.spongepowered.configurate.CommentedConfigurationNode

class VariableLevellingStrategy(
    minLevel: Int,
    maxLevel: Int
) : LevellingStrategy("Variable", minLevel, maxLevel) {
    override fun generate(context: Context): Int? {
        TODO("Not yet implemented")
    }

    override fun replaceInFormula(formula: String, context: Context): String {
        TODO("Not yet implemented")
    }

    companion object{
        fun parse(
            node: CommentedConfigurationNode
        ): VariableLevellingStrategy{
            TODO("Not yet implemented")
        }
    }
}